addons/share-through-mystuff/userscript.js

```diff
@@ -20,7 +20,31 @@ export default async function ({ addon, console, msg }) {
     event.preventDefault();
     const confirmation = await addon.tab.confirm(msg("confirmation-title"), msg("confirmation"));
     if (confirmation) {
-      event.target.parentElement.querySelector(".media-share").click();
+      const xToken = await addon.auth.fetchXToken();
+      const projectId = event.target
+        .closest(".media-item-content")
+        .querySelector(".media-thumb > a")
+        .href.match(/\d+/)[0];
+
+      const xhrOpen = XMLHttpRequest.prototype.open;
+      const xhrSend = XMLHttpRequest.prototype.send;
+      XMLHttpRequest.prototype.open = function (method, path, ...args) {
+        const newPath = `https://api.scratch.mit.edu/proxy/projects/${projectId}/share`;
+        xhrOpen.call(this, "PUT", newPath, ...args);
+        // CSRF token header is added by Scratch
+        this.setRequestHeader("x-token", xToken);
+        this.withCredentials = true;
+        this.send = () => {
+          this.setRequestHeader("X-Requested-With", ""); // Do not send this header
+          xhrSend.call(this); // Send empty body
+        };
+        return undefined;
+      };
+
+      event.target.parentElement.querySelector(".media-share").click(); // .click() is synchronous
+      // By this point, the request has already been sent. Remove traps.
+      XMLHttpRequest.prototype.open = xhrOpen;
+      XMLHttpRequest.prototype.send = xhrSend;
```

Comment:
This line (47) is unnecessary, as prototype.send is never changed. The `send` property of an object is changed in line 37, not of the prototype (or "class")